docs/do-not-merge.md

```diff
@@ -0,0 +1,464 @@
+
+For more information about this feature, see the following:
+
+-  [Type-aware schema discovery](https://druid.apache.org/docs/latest/ingestion/schema-design.html#type-aware-schema-discovery)
+-  [26.0.0 release notes for Schema auto-discovery](https://github.com/apache/druid/releases#26.0.0-highlights-auto-type-column-schema-%28experimental%29-schema-auto-discovery-%28experimental%29).
+- [26.0.0 release notes for array column types](https://github.com/apache/druid/releases#26.0.0-highlights-auto-type-column-schema-%28experimental%29).
+
+## Smart segment loading
+
+The Coordinator is now more stable and user-friendly and has the following behavior updates:
+
+- New segments that are underreplicated are now prioritized when you use the new `smartSegmentLoading` mode, which is enabled by default. Previously, it prioritized segments equally.
+- Items in segment load queues can be prioritized or canceled, improving reaction time for changes in the cluster and segment assignment decisions.
+- Leadership changes have less impact now, and the Coordinator doesn't get stuck even if re-election happens while a Coordinator run is in progress.
+
+Lastly, the `cost` balancer strategy performs much better now and is capable of moving more segments in a single Coordinator run. These improvements were made by borrowing ideas from the `cachingCost` strategy.
+
+For more information, see the following:
+
+- [Upgrade note for config changes related to smart segment loading](#segment-loading-config-changes)
+- [New coordinator metrics](#new-coordinator-metrics)
+- [Smart segment loading documentation](https://druid.apache.org/docs/latest/configuration/index.html#smart-segment-loading)
+
+[#13197](https://github.com/apache/druid/pull/13197) [#14385](https://github.com/apache/druid/pull/14385) [#14484](https://github.com/apache/druid/pull/14484)
+
+### Guardrail for subquery results
+
+Users can now add a guardrail to prevent subquery’s results from exceeding the set number of bytes by setting `druid.server.http.maxSubqueryRows` in the Broker's config or `maxSubqueryRows` in the query context. This guardrail is recommended over row-based limiting.
+
+This feature is experimental for now and defaults back to row-based limiting in case it fails to get the accurate size of the results consumed by the query.
+
+[#13952](https://github.com/apache/druid/pull/13952)
+
+### Added a new OSHI system monitor
+
+Added a new OSHI system monitor (`OshiSysMonitor`) to replace `SysMonitor`. The new monitor has a wider support for different machine architectures including ARM instances. We recommend switching to the new monitor. `SysMonitor` is now deprecated and will be removed in future releases.
+
+[#14359](https://github.com/apache/druid/pull/14359)
+
+
+## Java 17 support
+
+Druid now fully supports Java 17.
+
+[#14384](https://github.com/apache/druid/pull/14384)
+
+## Hadoop 2 deprecated
+
+Support for Hadoop 2 is now deprecated. It will be removed in a future release.
```

Comment:
```suggestion
Many of the important dependent libraries that Druid uses no longer supports Hadoop 2. In order for us to stay current and have pathways to mitigate security vulnerabilities, the community has decided to deprecate support for Hadoop 2.x releases starting this release. Starting Druid 28.x, Hadoop 3.x is the only supported Hadoop version.
Please consider migrating to MSQ based ingestion, native ingestion if you are using Hadoop 2.x for ingestion today. If migrating to Druid ingestion is not a possibility, you need to plan to upgrade your Hadoop infrastructure before upgrading to the next Druid release.
```

Reply:
Added manually. I left the highlight as this single line and a link to all this info in the Upgrade notes section 

https://github.com/apache/druid/pull/14600/commits/7b69c7c38ee92a8ed2fd23bc1e03caafa4856cd0